modules/host/announce.go

```diff
@@ -67,3 +59,28 @@ func (h *Host) Announce() error {
 
 	return nil
 }
+
+// Announce creates a host announcement transaction, adding information to the
+// arbitrary data, signing the transaction, and submitting it to the
+// transaction pool.
+func (h *Host) Announce() error {
+	// check that our address is reachable
+	lockID := h.mu.RLock()
+	addr := h.myAddr
+	h.mu.RUnlock(lockID)
+	if addr.Host() == "" {
+		return errors.New("can't announce without knowing external IP")
```

Comment:
I think I want to defer to --force means "do it anyway", simply because you might be on a testnet or you might be curious or etc. etc. etc. So we should leave any sort of check in Announce instead of announce.